integration_test/run_tests.sh

```diff
@@ -87,16 +133,32 @@ function cleanup {
   rm -rf $DIR/functions/node_modules/firebase-functions
 }
 
-build_sdk
-pick_node8
-install_deps
-delete_all_functions
-announce "Deploying functions to Node 8 runtime ..."
-deploy
-run_tests
-pick_node6
-announce "Re-deploying the same functions to Node 6 runtime ..."
-deploy
-run_tests
+if [[ $PROJECT_ID_NODE_6 == $PROJECT_ID_NODE_8 ]]; then 
+  build_sdk
+  pick_node8
+  install_deps
+  delete_all_functions
+  announce "Deploying functions to Node 8 runtime ..."
+  deploy
+  waitForPropagation
+  run_tests
+  pick_node6
+  announce "Re-deploying the same functions to Node 6 runtime ..."
+  deploy
+  waitForPropagation
+  run_tests
+else 
+  build_sdk
+  pick_node8
+  install_deps
+  delete_all_functions
+  announce "Deploying functions to Node 8 runtime ..."
+  deploy
+  pick_node6
+  announce "Re-deploying the same functions to Node 6 runtime ..."
+  deploy
+  waitForPropagation
+  run_all_tests
+fi
```

Comment:
For these lines:

https://github.com/firebase/firebase-functions/blob/baa48d5a72c1ee7bf532c6b9078673beda130046/integration_test/run_tests.sh#L136-L162

Can we remove the overall if/else and replace with something like this:

```suggestion
build_sdk
pick_node8
install_deps
delete_all_functions
announce "Deploying functions to Node 8 runtime ..."
deploy
if [[ $PROJECT_ID_NODE_6 == $PROJECT_ID_NODE_8 ]]; then
  waitForPropagation
  run_tests
fi
pick_node6
announce "Re-deploying the same functions to Node 6 runtime ..."
deploy
waitForPropagation
if [[ $PROJECT_ID_NODE_6 == $PROJECT_ID_NODE_8 ]]; then
  run_tests
else
  run_all_tests
fi
```